experimental/LieAlgebras/src/simple_lie_algebra.jl

```diff
@@ -0,0 +1,263 @@
+
+function ==(L::SimpleLieAlgebra{T}, M::SimpleLieAlgebra{T}) where T <: RingElement
+  (L.root_system).root_system_type != (M.root_system).root_system_type && error("Incompatible Root systems")
+  return L.mat_space == M.mat_space
+end
+
+function ==(f::SimpleLieAlgebraElem{T}, g::SimpleLieAlgebraElem{T}) where T <: RingElement
+  parent(f) != parent(g) && error("Incompatible Lie algebras")
+  return matrix(f) == matrix(g)
+end
+###############################################################################
+#
+#   Constructors
+#
+###############################################################################
+
+@doc raw"""
+    zero(L::SimpleLieAlgebra{T}) -> SimpleLieAlgebraElem{T}
+
+Return the zero element of the Lie algebra `L`.
+"""
+function zero(L::SimpleLieAlgebra{T}) where {T <: RingElement} 
+  x = zero(L.mat_space)
+  f = SimpleLieAlgebraElem{T}(L, x)
+  return f
+end
+
+function (L::SimpleLieAlgebra{T})() where {T <: RingElement}
+  return zero(L)
+end
+
+function (L::SimpleLieAlgebra{T})(A::MatElem{T}) where {T <: RingElem}
+  f = SimpleLieAlgebraElem{T}(L, L.mat_space(A))
+  return f
+end
+
+@doc raw"""
+    lie_algebra(R::Ring, S::String, cached::Bool=true) -> SimpleLieAlgebra{elem_type(R)}
+
+Construct the simple Lie algebra over the ring `R` with root system of type `S`
+"""
+function lie_algebra(R::Ring, S::String, cached::Bool=true)
+  T = elem_type(R)
+  return SimpleLieAlgebra{T}(R, S, cached)
+end
+
+###############################################################################
+#
+#   Chevalley basis and adjoint matrix
+#
+###############################################################################
+
+@doc raw"""
+    chevalley_basis(L::SimpleLieAlgebra{T}) -> Vector{Vector{SimpleLieAlgebraElem{T}}}
+
+Give the Chevalley basis of the simple Lie algebra `L`
+"""
+function chevalley_basis(L::SimpleLieAlgebra)
+  RS = root_system(L)
+  Rp = RS.positive_roots
+  sR = RS.simple_roots
+  n = length(Rp)
+  m = length(sR)
+  #root vectors
+  e1 = elem_type(L)[]
+  e2 = elem_type(L)[]
+  for i = 1:n
+    z1 = L()
+    z1[i] = 1
+    z2 = L()
+    z2[i+n] = 1
+    e1 = reduce(vcat, (e1, z1))
+    e2 = reduce(vcat, (e2, z2))
+  end
+  #basis for cartan algebra
+  e3 = elem_type(L)[]
+  for i = 1:m
+    z = L()
+    z[i+2*n] = 1
+    e3 = reduce(vcat, (e3, z))
+  end
+  return [e1, e2, e3]
+end 
+
+@doc raw"""
+    adjoint_matrix(L::SimpleLieAlgebra{T}) -> Vector{Any}
+
+Give the adjoint matrices of all basis vectors acting on the Lie algebra `L` with respect 
+to the Chevalley basis of `L`
+"""
+function adjoint_matrix(L::SimpleLieAlgebra{T}) where T <: RingElement #computes the adjoint matrix with respect to the Chevalley basis.
+  St = L.root_type
+  R = base_ring(L)	
+  n = St[2]
+  Q = GAP.Globals.Rationals
+  S = GAP.Obj(St[1])
+  LG = GAP.Globals.SimpleLieAlgebra(S, n, Q) #define the Lie algebra in Gap over the rationals, we are interested in the structure constants
```

Comment:
It would be great if this would work without creating another GAP Lie algebra, but not a blocker.

Reply:
fixed this so it now uses the structure constants via `bracket`. However,  I now define another Lie algebra to get the structure constants, so I'm not sure if this is better in terms of defining GAP Lie algebras.